app/models/project.rb

```diff
@@ -163,6 +163,20 @@ def remove_keyword! name, user
                    project: self)
   end
 
+  def similar_projects
+    if self.keywords.empty?
+      return []
+    end
+
+    similar = []
+    self.keywords.each do |keyword|
```

Comment:
What about changing the approach? Instead of going through the `keywords` and take the `projects`. You could take all the projects that are active and have those keywords.